raiden/tests/unit/test_tokennetwork.py

```diff
@@ -780,3 +780,214 @@ def test_routeing_issue2663(
+    #  |                    |
+    #  v                    v
+    # (2)  ----- x ----->  (3)
+    #  |                    |
+    #  |                    |
+    #  x                    x
+    #  |                    |
+    #  v                    v
+    # (4)                  (4)
+
+    channel_state1 = factories.make_channel(
+        our_balance=1,
+        our_address=our_address,
+        partner_balance=1,
+        partner_address=address1,
+    )
+    channel_state2 = factories.make_channel(
+        our_balance=2,
+        our_address=our_address,
+        partner_balance=0,
+        partner_address=address2,
+    )
+
+    # create new channels as participant
+    channel_new_state_change1 = ContractReceiveChannelNew(
+        transaction_hash=factories.make_transaction_hash(),
+        token_network_identifier=token_network_state.address,
+        channel_state=channel_state1,
+        block_number=open_block_number,
+    )
+    channel_new_state_change2 = ContractReceiveChannelNew(
+        transaction_hash=factories.make_transaction_hash(),
+        token_network_identifier=token_network_state.address,
+        channel_state=channel_state2,
+        block_number=open_block_number,
+    )
+
+    channel_new_iteration1 = token_network.state_transition(
+        payment_network_identifier=payment_network_state.address,
+        token_network_state=token_network_state,
+        state_change=channel_new_state_change1,
+        pseudo_random_generator=pseudo_random_generator,
+        block_number=open_block_number,
+    )
+
+    channel_new_iteration2 = token_network.state_transition(
+        payment_network_identifier=payment_network_state.address,
+        token_network_state=channel_new_iteration1.new_state,
+        state_change=channel_new_state_change2,
+        pseudo_random_generator=pseudo_random_generator,
+        block_number=open_block_number,
+    )
+
+    # create new channels without being participant
+    channel_new_state_change3 = ContractReceiveRouteNew(
+        transaction_hash=factories.make_transaction_hash(),
+        token_network_identifier=token_network_state.address,
+        channel_identifier=3,
+        participant1=address2,
+        participant2=address3,
+        block_number=open_block_number,
+    )
+
+    channel_new_iteration3 = token_network.state_transition(
+        payment_network_identifier=payment_network_state.address,
+        token_network_state=channel_new_iteration2.new_state,
+        state_change=channel_new_state_change3,
+        pseudo_random_generator=pseudo_random_generator,
+        block_number=open_block_number + 10,
+    )
+
+    channel_new_state_change4 = ContractReceiveRouteNew(
+        transaction_hash=factories.make_transaction_hash(),
+        token_network_identifier=token_network_state.address,
+        channel_identifier=4,
+        participant1=address3,
+        participant2=address1,
+        block_number=open_block_number,
+    )
+    token_network.state_transition(
```

Comment:
you should use the result of this state transition for the next state transitions as done in line 867. Same is true for the state changes below.